portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -26,23 +27,56 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
-import com.google.sps.data.Message;
 import java.util.ArrayList;
 
+/** An item containing visitor information and comment. */
+final class Message {
+    
+  public Message(long id, String name, String job, String comment, long timestamp) {
+    this.id = id;
+    this.name = name;
+    this.job = job;
+    this.comment = comment;
+    this.timestamp = timestamp;
+  }
+
+  private final long id;
+  private final String name;
+  private final String job;
+  private final String comment;
+  private final long timestamp;
+}
+
 /** Servlet that returns comments. */
 @WebServlet("/list-data")
 public class ListCommentsServlet extends HttpServlet {
-    
+
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Define a query rule that prioritizes latest messages
-    Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
- 
+    final Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING);
+    final PreparedQuery results = datastore.prepare(query);
+
+    // Get comment limit and check validity.
+    int commentLimit = 0;
+    try {
+      commentLimit = Integer.parseInt(request.getParameter("limit"));
```

Comment:
getParameter will return null if not set, which will cause failure. I guess we should provide default value for this case.

And invalid commentLimit should also be caught in the frontend.

Reply:
oh yes, catching it in the frontend will be much better. I'll fix this.